File: client/bcephal-client-model/Archives/ArchiveLogBrowserData.cs
using Bcephal.Models.Base;
using Newtonsoft.Json;

namespace Bcephal.Models.Archives
{
    public class ArchiveLogBrowserData : BrowserData
    {

        public string Username { get; set; }

        public string Message { get; set; }

        public long LineCount { get; set; }

        public string Status { get; set; }
        [JsonIgnore]
        public ArchiveLogStatus LogStatus
        {
            get { return !string.IsNullOrEmpty(Status) ? ArchiveLogStatus.GetByCode(Status) : null; }
            set { this.Status = value != null ? value.code : null; }
        }

        public string Action { get; set; }
        [JsonIgnore]
        public ArchiveLogAction LogAction
        {
            get { return !string.IsNullOrEmpty(Action) ? ArchiveLogAction.GetByCode(Action) : null; }
            set { this.Action = value != null ? value.code : null; }
        }

    }
}

File: server/bcephal-archive-service/src/main/java/com/moriset/bcephal/archive/domain/ArchiveLogBrowserData.java
/**
 * 
 */
package com.moriset.bcephal.archive.domain;

import com.moriset.bcephal.domain.BrowserData;

import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * @author MORISET-004
 *
 */
@Data
@EqualsAndHashCode(callSuper = false)
public class ArchiveLogBrowserData extends BrowserData {
	
	private String userName;

	private String message;

	private long lineCount;

	private ArchiveLogStatus status;

	private ArchiveLogAction action;

	public ArchiveLogBrowserData(ArchiveLog item) {
		super(item.getId(), item.getName());
		this.message = item.getMessage();
		this.userName = item.getUserName();
		this.action = item.getAction();
		this.status = item.getStatus();
		this.lineCount = item.getLineCount();
	}
}
